src/main/java/io/swagger/oas/inflector/utils/ExtensionsUtil.java

```diff
@@ -0,0 +1,170 @@
+package io.swagger.oas.inflector.utils;
+
+import io.swagger.oas.inflector.Constants;
+
+import io.swagger.oas.models.OpenAPI;
+import io.swagger.oas.models.Operation;
+import io.swagger.oas.models.PathItem;
+import io.swagger.oas.models.callbacks.Callback;
+import io.swagger.oas.models.media.ArraySchema;
+import io.swagger.oas.models.media.ComposedSchema;
+import io.swagger.oas.models.media.MediaType;
+import io.swagger.oas.models.media.Schema;
+import io.swagger.oas.models.parameters.Parameter;
+import io.swagger.oas.models.responses.ApiResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Set;
+
+public class ExtensionsUtil {
+   // private static final Logger LOGGER = LoggerFactory.getLogger(ResolverUtil.class);
+
+    private Map<String, Schema> schemas;
+
+
+
+
+    public void addExtensions(OpenAPI openAPI) {
+
+        if (openAPI.getComponents().getSchemas() != null) {
+            schemas = openAPI.getComponents().getSchemas();
+            if (schemas == null) {
+                schemas = new HashMap<>();
+            }
+        }
+
+
+        for (String name : schemas.keySet()) {
+            Schema schema = schemas.get(name);
+            if (schema.getExtensions() != null) {
+                if (!schema.getExtensions().containsKey(Constants.X_SWAGGER_ROUTER_MODEL)) {
+                    schema.addExtension(Constants.X_SWAGGER_ROUTER_MODEL, name);
+                }
+            } else {
+                schema.addExtension(Constants.X_SWAGGER_ROUTER_MODEL, name);
+            }
+        }
+        if(openAPI.getPaths() != null) {
+            for (String pathname : openAPI.getPaths().keySet()) {
+                PathItem pathItem = openAPI.getPaths().get(pathname);
+                resolvePath(pathItem);
+            }
+        }
+    }
+
+     public void resolvePath(PathItem pathItem){
+        for(Operation op : pathItem.readOperations()) {
```

Comment:
Please, replace "op" for a meaningful variable name. I think "op" is a brand for shoes :)